python/lsst/obs/base/ingest.py

```diff
@@ -249,6 +280,7 @@ def extractMetadata(self, filename: str) -> RawFileData:
             datasets = []
             FormatterClass = Formatter
             instrument = None
+            self._on_metadata_failure(filename, e)
```

Comment:
We probably should mention that this triggers before failFast logic triggers in the description above. Maybe this is telling us that we should remove the failFast logic completely some time in the future. Something for @mxk62 to ponder.

Reply:
I _do_ like `failFast` option as it allows me to get really nice and informative messages, practically effortlessly. Here's an example from the recent Gen3 ingestion of NTS Comcam data:
 ```
 status  |                                      error                                       | count 
---------+----------------------------------------------------------------------------------+-------
 SUCCESS |                                                                                  |  8769
 FAILURE | KeyError: "Could not find ['RAFTBAY'] in header"                                 |   135
 FAILURE | ValueError: None of the registered translation classes ['DECam', 'SuprimeCam', ' |    27
 ```
(Note that I "cut" the error message column to fit the limited space here; the table keeps error messages in their entirety.)
 
However, if the new error handling mechanism will allow me to access these pieces of information without much hassle, I won't have any fundamental reasons to be against removing `failFast` option.

Reply:
I think the big question is whether you ever want to ingest more than one file at a time; if so, I think the new approach would let you do that (which might be more efficient than single-file ingests) while getting exactly the same per-file error information back.  It will also let you get that information with single-file ingests, but `failFast` does that already.

Reply:
The automated ingesters were written to ingest a single file at a time merely because that was the only way to get a useful error message back from the code without parsing logs after the fact.  The ingester could easily be changed to call ingest with up to some max X files per RawIngestTask call if those calls return enough information to determine success or why it failed for **each** file.  Note, at this time there is no science knowledge built into the DBB BM such that it would know to group by exposure or wait until have all of the files for the exposure.  Current grouping would just be a limit on files in some arbitrary order.